argo-lite/src/services/AlgorithmUtils.js

```diff
@@ -44,4 +45,24 @@ export function averageClusteringCoefficient(snapshot) {
     const result = jsnx.averageClustering(jsnxGraph);
     console.log('Computing Clustering Coefficient');
     return result;
-}
\ No newline at end of file
+}
+
+/**
+ * Calculate the number of connected components in a graph
+ * @param {*} rawGraph the rawGraph inside appState
+ */
+export function connectedComponents(rawGraph) {
```

Comment:
Let's unify our API to pass in snapshot rather than rawGraph, since sometimes certain metadata might be needed. See `averageClusteringCoefficient` for example.

Reply:
Fixed in a new commit.